src/ddceler/DDcelerTMI.cc

```diff
@@ -0,0 +1,188 @@
+
+        // Create evaluator for parsing expressions with units
+        dd4hep::tools::Evaluator eval;
+
+        auto get_param
+            = [&](std::string const& key, double default_val) -> double {
+            if (!tracking_props.count(key))
+                return default_val;
+
+            // Values from RUNNER.field can include units (e.g., "0.025*mm" or
+            // "0.025")
+            std::string const& value_str = tracking_props.at(key);
+
+            // Evaluate expression to get value in DD4hep internal units (mm=1)
+            auto result = eval.evaluate(value_str.c_str());
+            CELER_VALIDATE(result.first == dd4hep::tools::Evaluator::OK,
+                           << "Failed to parse field tracking parameter '"
+                           << key << "' with value '" << value_str << "'");
+
+            // result.second is already in mm (DD4hep's base unit)
+            // Convert to Celeritas internal units
+            return result.second * celer_mm;
+        };
+
+        min_step = get_param("min_chord_step", min_step);
+        delta_chord = get_param("delta_chord", delta_chord);
+        delta_intersection
+            = get_param("delta_intersection", delta_intersection);
+    }
+
+    // Print field driver options
+    CELER_LOG(debug)
+        << "Field driver options: min_step=" << min_step / celer_mm
+        << " mm, delta_chord=" << delta_chord / celer_mm
+        << " mm, delta_intersection=" << delta_intersection / celer_mm
+        << " mm";
+
+    // Use a uniform magnetic field based on DD4hep ConstantField
+    auto make_field_input
+        = [field_direction, min_step, delta_chord, delta_intersection] {
+              celeritas::inp::UniformField input;
+
+              // Convert from DD4hep units (tesla) to Celeritas field units
+              constexpr double dd4hep_tesla = dd4hep::tesla;
+              input.strength = {field_direction.X() / dd4hep_tesla,
+                                field_direction.Y() / dd4hep_tesla,
+                                field_direction.Z() / dd4hep_tesla};
+
+              input.driver_options.minimum_step = min_step;
+              input.driver_options.delta_chord = delta_chord;
+              input.driver_options.delta_intersection = delta_intersection;
+              return input;
+          };
+    opts.make_along_step = celeritas::UniformAlongStepFactory(make_field_input);
+    opts.sd.ignore_zero_deposition = false;
+
+    // Save diagnostic file to a unique name
+    opts.output_file = "trackingmanager-offload.out.json";
+    return opts;
+}
+//---------------------------------------------------------------------------//
+
+void DDcelerTMI::constructPhysics(G4VModularPhysicsList* physics)
+{
+    CELER_LOG(info) << "Using Celeritas tracking for e-/e+/gamma.";
+
+    // Register Celeritas tracking manager
+    auto& tmi = TMI::Instance();
+    physics->RegisterPhysics(new celeritas::TrackingManagerConstructor(&tmi));
```

Comment:
Since this can throw, you may want to write (after including `accel/ExceptionConverter.hh`):

```suggestion
    CELER_TRY_HANDLE(
        physics->RegisterPhysics(new celeritas::TrackingManagerConstructor(&tmi)),
        ExceptionConverter{"ddceler.construct-physics"});
```

and same below with your call to `SetOptions(makeOptions())`